### ISSUE TITLE
`projects: open remote` selection requires the mouse

### ISSUE THREAD

**Issue by piwicode** [zed-industries/zed#28071]
### Summary

`project: open remote` command opens a modal dialog that cannot be navigated with the keyboard only.

Steps to trigger the problem:
1. Install & Open Zed
2. CTRL + SHIFT + P
3. type `projects: open remote`. A modal dialog named `Remote Projects (beta)` shows up.
4. hit `connect new server` and enter `ssh {user_name}@localhost`
5. hit enter to return to remote project modal
6. try to press the down or up keyboard arrow to navigate

Actual Behavior:

Nothing happens. The modal is no longer focused. This prevent to pick an option with the keyboard. One have to use the mouse instead.

Expected Behavior:

Change the selection of the proposed row of the modal dialog. We should be able to validate the selection with `Enter`.

### Zed Version and System Specs

Zed: v0.180.2 (Zed) 
OS: Linux Wayland debian unknown
Memory: 31.1 GiB
Architecture: x86_64
GPU: Intel(R) Xe Graphics (TGL GT2) || Intel open-source Mesa driver || Mesa 24.2.8-1

**Comment by lesha-co** [comment]
I want to add some details.
On my side bug happens when I go <kbd>cmd+shift+P</kbd> -> `projects: open remote` -> `connect new server` -> (command) -> <kbd>enter</kbd>

I can work around it by pressing <kbd>esc</kbd> and opening Remote Projects again with server already added.

Seems like after <kbd>enter</kbd> keyboard focus isn't returning to Remote Projects window

Zed: v0.180.2 (Zed) 
OS: macOS 15.4.0
Memory: 36 GiB
Architecture: aarch64

**Comment by probably-neb** [comment]
I believe this was fixed by #19851, as I am currently unable to reproduce this and can navigate the remote projects modal with arrow keys as well as <kbd>tab</kbd> + <kbd>shift-tab</kbd> as expected. Feel free to @-mention me and we can re-open if the issue is still occurring
